src/components/ClientPage/index.tsx

```diff
@@ -0,0 +1,159 @@
+        searchParams.set('search', queryPrm);
+        setSearchParams(searchParams);
+    }
+
+    function deleteQueryStrings() {
+        searchParams.delete('search');
+
+        setSearchParams(searchParams);
+    }
+
+    function getQueryStringAndFilterCompanies() {
+        const currentQueryString = searchParams.get('search');
+        const filteredCompanies = initialStateCompanies.filter((company) => {
+            if (
+                company.from.toLowerCase() ===
+                    currentQueryString?.toLowerCase() ||
+                company.companyName.toLowerCase() ===
+                    currentQueryString?.toLowerCase()
+            ) {
+                return true;
+            }
+        });
+        setCompanies(filteredCompanies);
+    }
+
+    const handleInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setInput(e.target.value);
+    };
+
+    const handleInputSearch = (e: React.SyntheticEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        if (input) {
+            setQueryStrings(input);
+            getQueryStringAndFilterCompanies();
+        } else {
+            deleteQueryStrings();
+            setCompanies(initialStateCompanies);
+        }
+
+        setInput('');
+    };
+
+    const handleButtonClick = (name: string) => {
+        searchParams.set(name.toLowerCase(), name.toLowerCase());
+        setSearchParams(searchParams);
+    };
+
+    const handleRemoveBtn = (name: string) => {
+        searchParams.delete(name.toLowerCase());
+        setSearchParams(searchParams);
+    };
+
+    function countButton() {
+        const filters: string[] = [];
+        const keys = searchParams.keys();
+        for (const key of keys) {
+            key !== 'search' ? filters.push(key) : null;
+        }
+        return filters.length;
+    }
+
+    return (
+        <section className={styles.clientPage}>
+            <h1>{t('description.clients')}</h1>
+
+            <div className={styles.inputField}>
+                <form onSubmit={handleInputSearch}>
+                    <input type="text" value={input} onChange={handleInput} />
+                    <button type="submit">
+                        <FontAwesomeIcon
+                            className={styles.icon}
+                            icon={faSearch}
+                        />
+                    </button>
+                </form>
+            </div>
+
+            <div className={styles.chooseCity}>
+                {continents.map((continent) => {
+                    const { id, name } = continent;
+                    return (
+                        <button
+                            key={id}
+                            onClick={() => handleButtonClick(name)}
+                        >
+                            {t(`description.${name.toLowerCase()}`)}
+                        </button>
+                    );
+                })}
+            </div>
+
+            <div className={styles.selectedCitys}>
```

Comment:
should be Cities

Reply:
tnx for the correction, i will change it!